src/main/java/com/andersen/hw/service/TicketServiceImpl.java

```diff
@@ -56,13 +67,24 @@ public void deleteById(Integer id) {
     }
 
     @Override
+    @Transactional
     public void updateTicket(Ticket ticket) {
         if (ticket == null) {
             throw new IllegalArgumentException("Ticket cannot be null");
         }
         ticketStorageDao.updateTicket(ticket);
     }
 
+    @Override
+    @Transactional
+    public void updateUserStatusAndCreateTicket(User user, Ticket ticket) {
+        if (updateUserFlag) {
+            userStorageDao.updateUserStatus(user);
+            ticketStorageDao.addTicket(ticket);
+        }
+        throw new IllegalArgumentException("The operation is disabled now");
+    }
```

Comment:
in this implementation exception will be thrown every time.  revert the logic:
```java
    public void updateUserStatusAndCreateTicket(User user, Ticket ticket) {
        if (!updateUserFlag) {
            throw new IllegalArgumentException("The operation is disabled now");
        }
        userStorageDao.updateUserStatus(user);
        ticketStorageDao.addTicket(ticket);
    }
```

Reply:
done